src/server/snapshot.cc

```diff
@@ -100,7 +114,13 @@ void SliceSnapshot::SerializeEntriesFb() {
     mu_.unlock();
 
     do {
-      PrimeTable::Cursor next = pt->Traverse(cursor, [this](auto it) { this->SaveCb(move(it)); });
+      if (cll) {
+        return;
+      }
+
+      PrimeTable::Cursor next = pt->Traverse(cursor, [this, &cll](auto it) {
+        this->SaveCb(move(it));
+      });
```

Comment:
We can (should?) check here for cancellation as well?

Reply:
i do not think so. this serializes into string and it's atomic.